woocommerce/woocommerce-ios
# Coupons: Add coupon details UI

Part of #5765 


### Description

This PR adds a new screen for coupon details. This screen should be navigated from the coupon list and the coupon search screen.

The screen is currently showing the main details of the coupon that is passed to the view model. Retrieving details from the API will be added in the next PR. The details include:
- Coupon code
- Coupon description
- Coupon discount: formatted based on discount type. I will have to confirm with the team how to format the amount for non-default discount types later.
- Coupon product limit: Currently only displaying All Products if there is no limit. Handling mapping product IDs to product names will be handled in another PR.
- Coupon expiry date

### Testing instructions

- Make sure that your test store has at least one coupon and the coupon management feature flag is enabled.
- Navigate to the Menu tab, select Coupons.
- In the Coupon list, select a coupon. Notice that the details are displayed correctly.
- Navigate back to the list, tap the Search icon and select a coupon from the Search screen. Notice that you are navigated the expected Coupon Details screen.

### Screenshots

| Portrait mode | Landscape mode | 
| ----- | ----- |
| <img src="https://user-images.githubusercontent.com/5533851/151495286-2cb4e006-2106-4c68-bcb1-5faabb7dfac5.png" width=320 /> | <img src="https://user-images.githubusercontent.com/5533851/151495325-8757c811-0bd3-48fc-bf10-6d27f46c9010.png" />


---
